manifests/charts/istio-control/istio-discovery/files/kube-gateway.yaml

```diff
@@ -55,6 +55,27 @@ spec:
       - name: istio-proxy
         image: "{{ .ProxyImage }}"
         {{with .Values.global.imagePullPolicy }}imagePullPolicy: "{{.}}"{{end}}
+        {{ if or (isset .Annotations "sidecar.istio.io/proxyCPU") (isset .Annotations "sidecar.istio.io/proxyMemory") (isset .Annotations "sidecar.istio.io/proxyCPULimit") (isset .Annotations "sidecar.istio.io/proxyMemoryLimit")  }}
```

Comment:
should we be handling the `.Values.global.proxy.resources` part? I have less concerns about setting that (reasonable defaults) since it doesn't tie us to an annotation API

Reply:
That is an interesting idea. It means we will have resource settings on gateway PODs independent of how the work on the `infrastructure` field progresses and later we can refine the logic here to respond to per-gateway resource settings from the `infrastructure` field.

Would you suggest I refactor this PR into using `.Values.global.proxy.resources` instead of `.Annotations` ?